### DISCUSSION TITLE
Is there any way to open a terminal in the main pane by default?

### DISCUSSION THREAD

**Discussion by dylf** [zed-industries/zed#14338] [Feature Requests]
Right now I'm running LazyGit as a task, rather than having it open in the button section of the editor I'd like the terminal window to open in the "Editor" area. I know you can drag the terminal up as a tab in the editor so the possibility is there. Is there any way to configure zed to do this by default?

Bonus points if we can auto-close the terminal buffer that opens when the task exits. :crossed_fingers: 

As a neovim user, I'm really enjoying the editor! :heart:

**Comment by SomeoneToIgnore** [comment]
There's no way, unofrunately — overall, terminal items in the main panel is something very unusual for Zed so far and no workflow (including the tasks) was accommodated to that.

**Comment by Tebro** [comment]
In the settings there is `terminal.dock` that can be set to `null` I was expecting this to open tasks and terminals as tabs in the current split pane, but unfortunately that wasn't the case. I would really like to request this as a feature, would bring me one step closer to my nvim workflow without having to drag the tabs out of the bottom terminal dock.

**Comment by SomeoneToIgnore** [comment]
https://github.com/zed-industries/zed/pull/22004 and https://github.com/zed-industries/zed/pull/22082 have added this feature.
Task docs https://zed.dev/docs/tasks#custom-keybindings-for-tasks now have an example of how things can be done using `"reveal_target": "center"` either in the task definition, or in the `task::Spawn` argument.

See also 
https://github.com/zed-industries/zed/blob/10792ee0ad3b58c24f9977c903d2b8a84dfbdd15/assets/settings/initial_tasks.json#L22-L25 
and
https://github.com/zed-industries/zed/blob/10792ee0ad3b58c24f9977c903d2b8a84dfbdd15/assets/keymaps/default-macos.json#L530-L532

**Reply by KaKi87** [reply]
I just tried on Mac : it doesn't show the same prompt as usual though (it shows `bash-3.2$` instead of `hostname:zed username$`) and doesn't execute `~/.bash_profile`.

**Comment by Tebro** [comment]
This is great! Thanks @SomeoneToIgnore for the guide. I made this task to get a terminal the same way, a bit yank but it works.

```json
  {
    "label": "terminal",
    "use_new_terminal": true,
    "allow_concurrent_runs": true,
    "shell": "system",
    "command": "$SHELL",
    "reveal_target": "center",
    "hide": "always"
  }
```
